olp-cpp-sdk-dataservice-read/src/repositories/PartitionsSaxHandler.cpp

```diff
@@ -0,0 +1,177 @@
+
+  return continue_parsing_;
+}
+
+bool PartitionsSaxHandler::String(const char* str, unsigned int length, bool) {
+  switch (state_) {
+    case State::kProcessingAttribute:
+      state_ = ProcessNextAttribute(str, length);
+      return continue_parsing_;
+
+    case State::kWaitForRootPartitions:
+      if (HashStringToInt("partitions") == HashStringToInt(str)) {
+        state_ = State::kWaitPartitionsArray;
+        return continue_parsing_;
+      } else {
+        return false;
+      }
+
+    case State::kParsingPartitionName:
+      partition_.SetPartition(std::string(str, length));
+      break;
+    case State::kParsingDataHandle:
+      partition_.SetDataHandle(std::string(str, length));
+      break;
+    case State::kParsingChecksum:
+      partition_.SetChecksum(std::string(str, length));
+      break;
+    case State::kParsingCrc:
+      partition_.SetCrc(std::string(str, length));
+      break;
+    case State::kParsingIgnoreAttribute:
+      break;
+
+    case State::kWaitForRootObject:     // Not expected
+    case State::kWaitForNextPartition:  // Not expected
+    case State::kWaitForRootObjectEnd:  // Not expected
+    case State::kParsingVersion:        // Version is not a string
+    case State::kParsingDataSize:       // DataSize is not a string
+    default:
+      return false;
+  }
+
+  state_ = State::kProcessingAttribute;
+
+  return continue_parsing_;
+}
+
+bool PartitionsSaxHandler::Uint(unsigned int value) {
+  if (state_ == State::kParsingVersion) {
+    partition_.SetVersion(value);
+  } else if (state_ == State::kParsingDataSize) {
+    partition_.SetDataSize(value);
+  } else {
+    return false;
+  }
+
+  state_ = State::kProcessingAttribute;
+  return continue_parsing_;
+}
+
+bool PartitionsSaxHandler::EndObject(unsigned int) {
+  if (state_ == State::kWaitForRootObjectEnd) {
+    state_ = State::kParsingComplete;
+    return true;  // complete
+  }
+
+  if (state_ != State::kProcessingAttribute) {
+    return false;
+  }
+
+  if (partition_.GetDataHandle().empty() || partition_.GetPartition().empty()) {
+    return false;  // partition is not valid
+  }
+
+  partition_callback_(std::move(partition_));
+
+  state_ = State::kWaitForNextPartition;
+
+  return continue_parsing_;
+}
+
+bool PartitionsSaxHandler::StartArray() {
+  // We expect only a single array in whol response
+  if (state_ != State::kWaitPartitionsArray) {
+    return false;
+  }
+
+  state_ = State::kWaitForNextPartition;
+
+  return continue_parsing_;
+}
+
+bool PartitionsSaxHandler::EndArray(unsigned int) {
+  if (state_ != State::kWaitForNextPartition) {
+    return false;
+  }
+
+  state_ = State::kWaitForRootObjectEnd;
+  return continue_parsing_;
+}
+
+bool PartitionsSaxHandler::Default() { return false; }
+
+void PartitionsSaxHandler::Abort() { continue_parsing_.store(true); }
+
+PartitionsSaxHandler::State PartitionsSaxHandler::ProcessNextAttribute(
+    const char* name, unsigned int length) {
+  switch (HashStringToInt(name)) {
```

Comment:
We could slightly improve it even further by introducing the fast-path for strings longer than 10 symbols (`dataHandle` is the longest one, and has 10 symbols). No need the hash those, just return `kParsingIgnoreAttribute`